koku/masu/database/ocp_report_db_accessor.py

```diff
@@ -715,6 +714,41 @@ def populate_monthly_tag_cost(
                     first_curr_month, first_next_month, cluster_id, cluster_alias, rate_type, rate_dict
                 )
 
+    def populate_monthly_tag_default_cost(
+        self, cost_type, rate_type, rate_dict, start_date, end_date, cluster_id, cluster_alias
+    ):
+        """
+        Populate the monthly default cost of a customer based on tag rates.
+
+        Right now this is just the node/month cost. Calculated from
+        tag value cost * number_unique_nodes for each tag key value pair
+        that is found on a line item with that node.
+        """
+        if isinstance(start_date, str):
+            start_date = parse(start_date).date()
+        if isinstance(end_date, str):
+            end_date = parse(end_date).date()
+
+        # usage_start, usage_end are date types
+        first_month = datetime.datetime(*start_date.replace(day=1).timetuple()[:3]).replace(tzinfo=pytz.UTC)
+        end_date = datetime.datetime(*end_date.timetuple()[:3]).replace(hour=23, minute=59, second=59, tzinfo=pytz.UTC)
+        # Calculate monthly cost for each month from start date to end date for each tag key:value pair in the rate
+        for curr_month in rrule(freq=MONTHLY, until=end_date, dtstart=first_month):
+            first_curr_month, first_next_month = month_date_range_tuple(curr_month)
+            LOG.info("Populating monthly tag based default cost from %s to %s.", first_curr_month, first_next_month)
+            if cost_type == "Node":
+                self.tag_upsert_monthly_default_node_cost_line_item(
+                    first_curr_month, first_next_month, cluster_id, cluster_alias, rate_type, rate_dict
+                )
+            elif cost_type == "Cluster":
+                self.tag_upsert_monthly_default_cluster_cost_line_item(
+                    first_curr_month, first_next_month, cluster_id, cluster_alias, rate_type, rate_dict
+                )
+            elif cost_type == "PVC":
+                self.tag_upsert_monthly_default_pvc_cost_line_item(
+                    first_curr_month, first_next_month, cluster_id, cluster_alias, rate_type, rate_dict
+                )
```

Comment:
I think including an else would be weird because we wouldn't know what metric it would be updating or where to update that metric. New metrics would just need to be added as additional statements in the if if we add new ones.

Reply:
That's fine. If there was a sensible default, that'd be idea. But if there isn't anything obvious, then we can skip it.

This is totally optional, but if you want to, you could add an else that logs a warning with the name of the unknown field. That way we can add an alert to watch for the message in the production logs. That'd at least help us queue up future changes without needing to wait for the customer to tell us something is amiss. :slightly_smiling_face: